openpype/hosts/blender/scripts/build_workfile.py

```diff
@@ -874,6 +875,111 @@ def build_lipsync(project_name: str, shot_name: str):
             )
 
 
+def build_fabrication(project_name, asset_name):
+    """Build fabrication workfile
+
+    Args:
+        project_name (str): The current project name from OpenPype Session.
+        asset_name (str): The current asset name from OpenPype Session.
+    """
+
+    # Get subsets
+    asset_doc = get_asset_by_name(
+        project_name, "LightSetupBank", fields=["_id"]
+    )
+    subsets = get_subsets(
+        project_name,
+        asset_ids={asset_doc["_id"]},
+        fields=["_id", "name", "data"],
+    )
+
+    representations = []
+    # Match and download and load subset
+    for subset in subsets:
+        raw_asset_name = asset_name.split("_")[0].lower()
+        raw_subset_name = (
+            subset["name"].strip("lighting").rsplit("_", 1)[0].lower()
+        )
+        if raw_asset_name == raw_subset_name:
+            representation = download_subset(
+                project_name,
+                "LightSetupBank",
+                subset["name"],
+            )
+            representations.append(representation)
+            load_subset(
+                project_name, representation, "AppendBlenderLightingLoader"
+            )
+    # Wait for downloads to be finished
+    wait_for_download(project_name, representations)
+
+    # Create setdress instance
+    bpy.ops.scene.create_openpype_instance(
+        creator_name="CreateWoollySetdress",
+        asset_name=asset_name,
+        subset_name="setdressMain",
+        gather_into_collection=True,
+    )
+
+    # Create camera instance
+    # If asset_name + '_cameraMain' exist don't create it
+    bpy.ops.scene.create_openpype_instance(
+        creator_name="CreateCamera",
+        asset_name=asset_name,
+        subset_name="cameraMain",
+        gather_into_collection=True,
+        datapath="objects",
+        datablock_name=asset_name +"_cameraMain",
+    )
+
+    # Create review instance
+    bpy.ops.scene.create_openpype_instance(
+        creator_name="CreateReview",
+        asset_name=asset_name,
+        subset_name="reviewMain",
+        datapath="collections",
+        datablock_name=asset_name + "_cameraMain",
+        use_selection=False,
+    )
+
+    # Set camera values
+    camera = bpy.data.objects[asset_name + "_cameraMain"]
+    camera.data.lens = 200
+    camera.location[1] = -10000
+    camera.rotation_euler[0] = -1.5708
+
+    # Create empty and rename it
+    bpy.ops.object.empty_add(
+        type="PLAIN_AXES", align="WORLD", location=(0, 0, 0), scale=(1, 1, 1)
+    )
+    bpy.data.objects["Empty"].name = "DOF_ctrl_object"
+
+    # Link DOF_ctrl to cameraMain collection
+    bpy.data.collections[asset_name + "_cameraMain"].objects.link(
+        bpy.data.objects["DOF_ctrl_object"]
+    )
+    # unlink DOF_ctrl from SceneCollection collection
+    bpy.context.scene.collection.objects.unlink(
+        bpy.data.objects["DOF_ctrl_object"]
+    )
```

Comment:
How about creating the empty in blender data, so it doesn't go in the scene collection, then you don't need to unlink it from your scene collection at all?